test/cppsuite/test_harness/workload/database_model.h

```diff
@@ -48,113 +54,177 @@ struct value_t {
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        std::vector<std::string> collection_names;
-
-        for (auto const &it : _collections)
-            collection_names.push_back(it.first);
-
-        return (collection_names);
+        _key_count += increment;
+        return (_key_count);
     }
 
     std::map<key_value_t, key_t>
-    get_keys(const std::string &collection_name)
+    get_keys()
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        return (_collections.at(collection_name).keys);
+        return (keys);
     }
 
     value_t
-    get_record(const std::string &collection_name, const char *key)
+    get_record(const char *key)
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        return (_collections.at(collection_name).values.at(key));
+        return (values.at(key));
     }
 
     void
-    insert_record(const std::string &collection_name, const char *key, const char *value)
+    insert_record(const char *key, const char *value)
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        auto &c = _collections.at(collection_name);
-        c.keys[key].exists = true;
+        keys[key].exists = true;
         value_t v;
         v.value = key_value_t(value);
-        c.values.emplace(key_value_t(key), v);
+        values.emplace(key_value_t(key), v);
     }
 
     void
-    update_record(const std::string &collection_name, const char *key, const char *value)
+    update_record(const char *key, const char *value)
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        auto &c = _collections.at(collection_name);
-        c.values.at(key).value = key_value_t(value);
+        values.at(key).value = key_value_t(value);
     }
 
     void
-    delete_record(const std::string &collection_name, const char *key)
+    delete_record(const char *key)
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        auto &c = _collections.at(collection_name);
-        c.keys.at(key).exists = false;
-        c.values.erase(key);
+        keys.at(key).exists = false;
     }
 
+    std::map<key_value_t, key_t> keys = {};
+    std::map<key_value_t, value_t> values = {};
+    const std::string name;
+    const uint64_t id;
+
     private:
-    /* Take a const id, not a reference as we're copying in an atomic. */
-    std::string
+    std::mutex _mtx;
+    uint64_t _key_count;
+};
+
+/* Representation of the collections in memory. */
+class database {
+    public:
+    /*
+     * Add a new collection, this will create the underlying collection in the database.
+     */
+    void
+    add_collection(uint64_t key_count = 0)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        if (_session == nullptr)
+            _session = connection_manager::instance().create_session();
+        uint64_t next_id = _next_collection_id++;
+        std::string collection_name = build_collection_name(next_id);
+        _collections.emplace(next_id, new collection(next_id, key_count, collection_name));
```

Comment:
_collections is defined as `std::map<uint64_t, std::shared_ptr<collection>>`, it should be std::shared_ptr instead of new

Reply:
Fixed in 8c23e74